yesod-test/Yesod/Test.hs

```diff
@@ -439,9 +439,10 @@ assertNoHeader header = withResponse $ \ SResponse { simpleHeaders = h } ->
 -- | Assert the last response is exactly equal to the given text. This is
 -- useful for testing API responses.
 bodyEquals :: HasCallStack => String -> YesodExample site ()
-bodyEquals text = withResponse $ \ res ->
-  liftIO $ HUnit.assertBool ("Expected body to equal " ++ text) $
-    (simpleBody res) == encodeUtf8 (TL.pack text)
+bodyEquals text = withResponse $ \ res -> do
+  let actual = simpleBody res
+      msg    = concat ["Expected body to equal ", text, ". Actual is: ", BSL8.unpack actual]
```

Comment:
Would it make sense for display purposes to put `text` and `actual` on their own lines? Also, I think it would be better to use UTF8 decoding (maybe via `decodeUtf8With lenientDecode`) instead of Char8.

Reply:
Changes made. Let me know what you think.